src/api/annotation/AnnotationAPI.js

```diff
@@ -266,16 +280,35 @@ export default class AnnotationAPI extends EventEmitter {
         return modelAddedToResults;
     }
 
+    #combineSameTargets(results) {
+        return results.reduce((accumulator, currentAnnotation) => {
+            const existingAnnotation = accumulator.find((annotationToFind) => {
+                return _.isEqual(currentAnnotation.targets, annotationToFind.targets);
+            });
+            if (!existingAnnotation) {
+                accumulator = [...accumulator, currentAnnotation];
```

Comment:
Depending on the size of the dataset, spreading the accumulator within a `reduce()` could have a significant performance impact. [Or so I've read](https://prateeksurana.me/blog/why-using-object-spread-with-reduce-bad-idea/). 

Something to consider if the size of `results` could be > 1000 elements.

Reply:
Ok, I'll see if we can do this another way

Reply:
Also, very nice catch!